opm/material/fluidmatrixinteractions/EclStone1Material.hpp

```diff
@@ -359,11 +359,11 @@ class EclStone1Material : public TraitsT
         // oil relperm at connate water saturations (with Sg=0)
         const Scalar krocw = params.krocw();
 
-        const Evaluation& Sw = Opm::decay<Evaluation>(fluidState.saturation(waterPhaseIdx));
-        const Evaluation& Sg = Opm::decay<Evaluation>(fluidState.saturation(gasPhaseIdx));
+        const Evaluation Sw = Opm::decay<Evaluation>(fluidState.saturation(waterPhaseIdx));
```

Comment:
This looks like a bug fix. What did Sw refer to here?

Reply:
> This looks like a bug fix.

More like a tiny optimisation.  `Opm::decay<>` constructs a new object and I recently became aware that there's no way for the compiler to perform lifetime extension (i.e., `const Evaluation& x = new_object`) without effectively doing something like this
```C++
const Evaluation __hidden = new_object;
const Evaluation& Sw = __hidden;
```
The above just bypasses the `__hidden` object.